xonsh/shell.py

```diff
@@ -48,6 +48,10 @@ def __init__(self, ctx=None, shell_type=None, config=None, rc=None,
         # shell type is pulled from env
         if shell_type is None:
             shell_type = env.get('SHELL_TYPE')
+            if shell_type == 'none':
```

Comment:
No, that's `shell_type is None`, as in unset.

This is to disallow the environment to leave xonsh in an unusable state.